nibabel/tests/test_openers.py

```diff
@@ -367,3 +370,126 @@ def test_iter():
         lobj = Opener(Lunk(''))
         with pytest.raises(TypeError):
             list(lobj)
+
+
+def md5sum(fname):
+    with open(fname, "rb") as fobj:
+        return hashlib.md5(fobj.read()).hexdigest()
+
+
+def test_DeterministicGzipFile():
+    with InTemporaryDirectory():
+        msg = b"Hello, I'd like to have an argument."
+
+        # No filename, no mtime
+        with open("ref.gz", "wb") as fobj:
+            with GzipFile(filename="", mode="wb", fileobj=fobj, mtime=0) as gzobj:
+                gzobj.write(msg)
+        anon_chksum = md5sum("ref.gz")
+
```

Comment:
minor: since file is small, and time is stored up to a second, shouldn't we sleep here for a second to 100% ensure that time is different between two invocations?

Reply:
This is checking changing the filename. We mock mtime on L412+

Reply:
great, thanks!